model-optimizer/extensions/ops/non_max_suppression.py

```diff
@@ -57,31 +55,56 @@ def backend_attrs(self):
 
     @staticmethod
     def infer(node: Node):
+        num_of_inputs = len(node.in_ports())
+        opset = node.get_opset()
+        max_num_of_inputs = 6 if opset == 'opset5' else 5
+        input_msg_fmt = 'NonMaxSuppression node {} from {} must have from 2 to {} inputs'
+        inputs_msg = input_msg_fmt.format(node.soft_get('name', node.id), opset, max_num_of_inputs)
+        assert 2 <= num_of_inputs <= max_num_of_inputs, inputs_msg
+
         boxes_shape = node.in_port(0).data.get_shape()
         assert boxes_shape is not None, 'The shape of tensor with boxes is not defined'
         scores_shape = node.in_port(1).data.get_shape()
         assert scores_shape is not None, 'The shape of tensor with scores is not defined'
         assert len(boxes_shape) == 3, 'Length of tensors with boxes must be equal to 3'
         assert len(scores_shape) == 3, 'Length of tensors with scores must be equal to 3'
 
-        max_output_boxes_per_class = node.in_port(2).data.get_value()
-        if max_output_boxes_per_class is None:
+        if num_of_inputs >= 3:
```

Comment:
Please, add a comment describing this check and why the `max_output_boxes_per_class = 0` for the `else` case

Reply:
Done.